web/src/message_edit.js

```diff
@@ -467,6 +476,12 @@ function edit_message($row, raw_content) {
         }),
     );
 
+    if (currently_echoing) {
+        $form.find(".message_edit_save_container").hide();
+    } else {
+        $form.find(".message_edit_saving_container").hide();
+    }
```

Comment:
Can you explain the mechanism through which the right one of these gets shown again if the current message-edit save completes?

Reply:
So like currently, if you click on Save, and you open the message edit then while the current message is echoing locally ("saving"), then the `Saving` button would appear as disabled. In case the save fails, then the `message_edit_saving_container` would be hidden, making the `Saving` button to be disabled while showing `Save`.

However, in case when the Save succeeds, then the entire message edit box closes despite showing the `Saving` button. Do you want the message edit box to remain open and rather show the "Save" button similar to that of when save fails?

Reply:
I'm not sure precisely what you're asking, but I think we indeed don't want to close the edit box if you're opened it and started working on a next edit to the message, just because the previous edit save completed.

Reply:
I'm not sure the right way to proceed -- the message container would be re rendered after the event to update message is received. So I guess I should open the edit box within the `message_list_view.rerender_messages` method?

Reply:
updated!

Reply:
While your change in `message_events` ensures the edit state doesn't close, but now there's no way the Saving button can hide and the Save button can be shown if the echoing completes when the next edit is being made.

Basically, we need a live update for the Saving / Save button but there isn't one.